### DISCUSSION TITLE
How to rename a existed vector name of a collection?

### DISCUSSION THREAD

**Discussion by failable** [qdrant/qdrant#2750] [Q&A] [answered]
The title says it all...

**Comment by timvisee** [comment, marked as answer]
At this time, Qdrant  does not support changing vector names. Nor does it support adding/removing vectors once the collection is created.

To achieve this you can recreate the collection with the new desired vector names.

**Reply by constegit** [reply]
Hello, is it still the case? Thanks

**Reply by timvisee** [reply]
Yes.

**Comment by nauyan** [comment]
@timvisee is this still the case?

**Reply by timvisee** [reply]
It is, sorry!

We do have plans to allow you to change this on existing collections, but I cannot give you an estimate on this yet.

**Reply by nauyan** [reply]
Got it. 

Thanks for the prompt response.

**Comment by LisaBaumann** [comment]
@timvisee is this still the case ?

**Reply by timvisee** [reply]
Yes. You cannot rename an existing vector at this time.

**Reply by DominiquePaul** [reply]
What about adding a new vector? :)

**Reply by timvisee** [reply]
Also not supported at this time, sorry!

When a new vector is needed we recommend to create a new collection from scratch.

**Reply by DominiquePaul** [reply]
Ok, thank you! In my case I've got a collection with vectors for over 5M images. I want to add a second set of vectors. Maybe a third in the future.

Would your recommendation be then that I write a script that copies/migrates the embeddings and payloads from one collection to the other and drops it then? Thanks!

**Reply by timvisee** [reply]
At this time, yes. Sorry for the inconvenience.

This is definitely still something we'd like to support - changing the vector schema on an existing collection. But we don't have it on our planning yet.

**Comment by jony89** [comment]
Any time soon to support this? mandatory feature
@timvisee

**Comment by cmohamma** [comment]
Is this still the case? I find it hard to recommend this product to others when it lacks such a basic functionality...

**Comment by dhirajsuvarna** [comment]
+ I am also looking for this feature

**Comment by rklf** [comment]
Any news on this please @timvisee ? If no, any plan to support this?

**Comment by whoisronak** [comment]
hi @timvisee. first off, thanks for the product. its great. this seems like a fairly critical feature. putting another comment here to perhaps bump the priority. thanks!

**Comment by barela14** [comment]
+ yeah that would be pretty useful
